pkg/controllers/controlplanemachinesetgenerator/controller_test.go

```diff
@@ -1870,3 +1871,585 @@ var _ = Describe("controlplanemachinesetgenerator controller on GCP", func() {
+			Zone: "az2",
+		})
+
+		az3ProviderSpecBuilderOpenStack = machinev1beta1resourcebuilder.OpenStackProviderSpec().WithZone("az3").WithRootVolume(&machinev1alpha1.RootVolume{
+			Zone: "az3",
+		})
+
+		az4ProviderSpecBuilderOpenStack = machinev1beta1resourcebuilder.OpenStackProviderSpec().WithZone("az4")
+
+		az5ProviderSpecBuilderOpenStack = machinev1beta1resourcebuilder.OpenStackProviderSpec().WithRootVolume(&machinev1alpha1.RootVolume{
+			Zone: "az5",
+		})
+
+		cpms3FailureDomainsBuilderOpenStack = machinev1resourcebuilder.OpenStackFailureDomains().WithFailureDomainBuilders(
+			az1FailureDomainBuilderOpenStack,
+			az2FailureDomainBuilderOpenStack,
+			az3FailureDomainBuilderOpenStack,
+		)
+
+		cpms5FailureDomainsBuilderOpenStack = machinev1resourcebuilder.OpenStackFailureDomains().WithFailureDomainBuilders(
+			az1FailureDomainBuilderOpenStack,
+			az2FailureDomainBuilderOpenStack,
+			az3FailureDomainBuilderOpenStack,
+			az4FailureDomainBuilderOpenStack,
+			az5FailureDomainBuilderOpenStack,
+		)
+
+		cpmsInactive3FDsBuilderOpenStack = machinev1resourcebuilder.ControlPlaneMachineSet().
+							WithState(machinev1.ControlPlaneMachineSetStateInactive).
+							WithMachineTemplateBuilder(
+				machinev1resourcebuilder.OpenShiftMachineV1Beta1Template().
+					WithProviderSpecBuilder(
+						az1ProviderSpecBuilderOpenStack.WithZone("").WithRootVolume(&machinev1alpha1.RootVolume{}).WithFlavor("m1.large"),
+					).
+					WithFailureDomainsBuilder(machinev1resourcebuilder.OpenStackFailureDomains().WithFailureDomainBuilders(
+						az1FailureDomainBuilderOpenStack,
+						az2FailureDomainBuilderOpenStack,
+						az3FailureDomainBuilderOpenStack,
+					)),
+			)
+
+		cpmsInactive5FDsBuilderOpenStack = machinev1resourcebuilder.ControlPlaneMachineSet().
+							WithState(machinev1.ControlPlaneMachineSetStateInactive).
+							WithMachineTemplateBuilder(
+				machinev1resourcebuilder.OpenShiftMachineV1Beta1Template().
+					WithProviderSpecBuilder(
+						az1ProviderSpecBuilderOpenStack.WithZone("").WithRootVolume(&machinev1alpha1.RootVolume{}).WithFlavor("m1.large"),
+					).
+					WithFailureDomainsBuilder(machinev1resourcebuilder.OpenStackFailureDomains().WithFailureDomainBuilders(
+						az1FailureDomainBuilderOpenStack,
+						az2FailureDomainBuilderOpenStack,
+						az3FailureDomainBuilderOpenStack,
+						az4FailureDomainBuilderOpenStack,
+						az5FailureDomainBuilderOpenStack,
+					)),
+			)
+
+		cpmsActiveOutdatedBuilderOpenStack = machinev1resourcebuilder.ControlPlaneMachineSet().
+							WithState(machinev1.ControlPlaneMachineSetStateActive).
+							WithMachineTemplateBuilder(
+				machinev1resourcebuilder.OpenShiftMachineV1Beta1Template().
+					WithProviderSpecBuilder(
+						az1ProviderSpecBuilderOpenStack.WithZone("").WithRootVolume(&machinev1alpha1.RootVolume{}).WithFlavor("m1.large"),
```

Comment:
Or whatever machine type you want, just different from `m1.large`.

```suggestion
						az1ProviderSpecBuilderOpenStack.WithZone("").WithRootVolume(&machinev1alpha1.RootVolume{}).WithFlavor("m1.xlarge"),
```